src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -421,6 +451,21 @@ protected void closeResources() {
     return null;
   }
 
+  private long getNextUpdateTime(String pollIntervalMode) {
+    if (JdbcSourceTaskConfig.POLL_INTERVAL_MODE_FIXED.equals(pollIntervalMode)) {
+      return config.getInt(JdbcSourceTaskConfig.POLL_INTERVAL_MS_CONFIG);
+    } else {
+      Optional<Duration> optionalDuration =
+          cronExecutionTime.timeToNextExecution(ZonedDateTime.now(ZoneOffset.UTC));
+      if (optionalDuration.isPresent()) {
+        return optionalDuration.get().toMillis();
+      } else {
+        log.warn("Cron expression provided does not define a next execution.");
+        return Duration.ofMinutes(10).toMillis();
```

Comment:
In that case you will execute the query every 10 ms, won't you?
What about throwing an exception?
If logic is correct, please describe why it works

Reply:
This piece of code would be triggered for example if the cron schedule would be something like "for every second in 2019". I guess it's rather unusal. Anyway, if the user wants to set it for some reason, I don't see why the connector should fail. It will stay running doing nothing (similarly to incremental mode and no changes in DB). The idea here was to return anything greater than 0. It doesn't really matter what as far as the code is concerned, it will sleep for at most 100ms and call this method again.

Reply:
Yes, but it is coupled with `querier.getLastUpdate()`:
```
final long nextUpdate = querier.getLastUpdate() + getNextUpdateTime() // here you return 10 for `getNextUpdateTime`
```
so next time you arrive in this function you have high chances to have `sleepMs <= 0`

Reply:
I think there was an issue there that I solved in the last refactoring.